src/main/resources/api/service_openapi.yaml

```diff
@@ -216,12 +263,90 @@ components:
             format: uuid
         jobControl:
           $ref: '#/components/schemas/JobControl'
+    CreateDataReferenceRequestBody:
+      type: object
+      properties:
+        name:
+          description: The name of the data reference
+          type: string
+        resourceId:
+          description: The ID of the resource
+          type: string
+          nullable: true
+        referenceType:
+          description: The type of the data reference (i.e. TDR snapshot, AWS bucket, etc)
+          type: string
+          nullable: true
+        reference:
+          description: The reference to the data (i.e. gs://...) #todo- describe as json instead
+          type: object
+          nullable: true
+        cloningInstructions:
+          description: The cloning instructions for the reference
+          type: string
+        credentialId:
+          description: The IDs of the credentials to use with this resource
+          type: string
+          nullable: true
+        jobControl:
+          $ref: '#/components/schemas/JobControl'
+    CreatedDataReference:
+      type: object
+      properties:
+        id:
+          type: string
+          description: UUID of the newly referenced data todo- add more detail
     CreatedWorkspace:
       type: object
       properties:
         id:
           type: string
           description: UUID of a newly-created workspace
+    DataReference:
+      type: object
+      properties:
+        referenceId:
+          description: The ID of the data reference
+          type: string
+          format: uuid
+        name:
+          description: The name of the data reference; used to refer to the reference
+          type: string
+        workspaceId:
+          description: The ID of the workspace containing the reference
+          type: string
+          format: uuid
+        resourceId:
+          description: The ID of the resource, if it is controlled by Workspace Manager
+          type: string
+          format: uuid
+          nullable: true
```

Comment:
Thoughts on making this something like a `ResourceDescription` object with fields from the workspace_resource table? It would still just be null for uncontrolled data resources, but it would have potentially useful information for controlled data resources.